### DISCUSSION TITLE
More flexible date ranges for Trend Metrics

### DISCUSSION THREAD

**Discussion by alies-dev** [laravel/nova-issues#6411] [Ideas & Feature Requests]
`\Laravel\Nova\Metrics\Trend::aggregate` method has lines:
```php
$startingDate = $this->getAggregateStartingDate($request, $unit, $timezone),
$endingDate = CarbonImmutable::now($timezone),
```

it will be nice to extract `CarbonImmutable::now($timezone)` into another protected method, similar to `getAggregateStartingDate`, so developers can override it for customisation purposes (I have to display trends for whole years, like 2023, 2022, etc, and do not end them by today's date).

**Comment by crynobone** [comment]
Released with Laravel Nova 5.0.0

Feel free to open up a new issue if you're still experiencing this problem on the latest version.
